src/github.com/couchbase/sync_gateway/db/kv_channel_storage.go

```diff
@@ -204,8 +242,30 @@ func (b *BitFlagStorage) writeSingleBlockWithCas(entries []*LogEntry) error {
 		return err
 	}
 	block.SetCas(casOut)
-
 	return nil
+
+}
+
+func (b *BitFlagStorage) marshalBlock(entries []*LogEntry) ([]byte, IndexBlock, error) {
+	if len(entries) == 0 {
+		return nil, nil, nil
+	}
+	// Get block based on first entry
+	block := b.getIndexBlockForEntry(entries[0])
+	// Apply updates to the in-memory block
+	for _, entry := range entries {
+		err := block.AddEntry(entry)
+		if err != nil { // Wrong block for this entry
+			return nil, nil, err
+		}
+	}
+	localValue, err := block.Marshal()
+	if err != nil {
+		base.Warn("Unable to marshal channel block - cancelling block update")
+		return nil, block, errors.New("Error marshalling channel block")
+	}
+	return localValue, block, nil
+
```

Comment:
It's part of this commit: https://github.com/couchbase/sync_gateway/commit/f416d4442accebceff8d713e88d087050c66a99c

which got lopped into this PR due to code / manual testing dependencies.